src/commands/update.rs

```diff
@@ -0,0 +1,236 @@
+      (false, true) => Self::Force,
+      (false, false) => Self::Regular,
+    }
+  }
+}
+
+pub fn invoke(args: &ArgMatches) {
+  info!("Checking for updates");
+
+  if !server::is_installed() {
+    error!("Failed to find server executable. Can't update!");
+    exit(1);
+  }
+
+  let dry_run = args.is_present("dry_run");
+  let check = args.is_present("check");
+  let force = args.is_present("force");
+
+  let server_running = server::is_running();
+  let update_available = update_is_available();
+  if update_available {
+    info!("A server update is available!");
+  } else {
+    info!("No server updates found");
+  }
+
+  match Action::new(check, force) {
+    Action::Check => match (dry_run, update_available) {
+      (true, true) => {
+        info!("Dry run: An update is available. This would exit with 0 to indicate this.")
+      }
+      (true, false) => {
+        info!("Dry run: No update is available. This would exit with 1 to indicate this.")
+      }
+      // 0 indicates that an update is available
+      (false, true) => exit(0),
+      // TODO: should we do a value other than 1 here, and if we do then what value?
+      // 1 indicates the server is up to date
+      (false, false) => exit(1),
+    },
+    Action::Force => match (dry_run, server_running) {
+      (true, true) => info!("Dry run: Server would be shutdown, updated, and brought back online"),
+      (true, false) => info!("Dry run: The server is offline and would be updated"),
+      _ => {
+        debug!("Force updating!");
+        update_server();
+      }
+    },
+    Action::Regular => {
+      if dry_run {
+        match (server_running, update_available) {
+          (true, true) => info!(
+            "Dry run: An update is available and the server is ONLINE. \
+                    The server would be shutdown, updated, and brought back online."
+          ),
+          (false, true) => info!(
+            "Dry run: An update is available and the server is OFFLINE. \
+                    The server would be updated."
+          ),
+          (_, false) => info!("Dry run: No update is available. Nothing to do."),
+        }
+      } else if update_available {
+        debug!("Updating the installation!");
+        update_server()
+      }
+    }
+  }
+}
+
+fn update_server() {
+  // Shutdown the server if it's running
+  let server_was_running = server::is_running();
+  if server_was_running {
+    server::send_shutdown();
+    server::wait_for_exit();
+  }
+
+  // Update the installation
+  if let Err(e) = server::install(constants::GAME_ID) {
+    error!("Failed to install server: {}", e);
+    exit(1);
+  }
+
+  // Bring the server up if it was running before
+  if server_was_running {
+    let config = load_config();
+    match server::start_daemonized(config) {
+      Ok(_) => info!("Server daemon started"),
+      Err(e) => {
+        error!("Error daemonizing: {}", e);
+        exit(1);
+      }
+    }
+  }
+}
+
+fn update_is_available() -> bool {
```

Comment:
1. Rename to `check_update_status` and return some sort of `UpdateInfo` instead that stores the different buildids as well as exposing an `update_available() -> bool` method. This exposes this information in a convenient way for anyone else that wants to use it.
2. Move these functions into `crate::server`
3. Add in unit tests for this (will require some `#[cfg(test)]` to modify the signature

Reply:
Still need to cover 3.